### DISCUSSION TITLE
Enabling Skew Protection on a statically rendered app

### DISCUSSION THREAD

**Discussion by bitttttten** [vercel/community#6496] [Help] [answered]
### Summary

Hey all,

We have a statically rendered app using Vite and we'd like to enable skew protection on it.

What's the recommended approach for this? 

> Append the value of process.env.VERCEL_DEPLOYMENT_ID to each request using one of the following:
> - dpl query string parameter
> - x-deployment-id header
> - __vdpl cookie

From the above note in the docs, it seems we'd need to set a cookie. Although I'm wondering what the best practise would be.

### Steps to Reproduce

n/a

**Comment by gajraj-gan** [comment]
Were you able to do it?

**Reply by bitttttten** [reply]
Yes, I wrote my own vite plugin for it. Do you want me to share it?

**Reply by gajraj-gan** [reply]
Yes, that would be awesome !

**Reply by bitttttten** [reply, marked as answer]
@gajraj-gan I published it to npm: https://github.com/bitttttten/vite-vlugin-vercel-skew-protection/

Do you want to give it a look and see if it works for you? There may be some edge cases I have missed, and you might be using a different vite.config than we are, however this works well for us!

**Reply by bitttttten** [reply]
I would recommend testing this plugin on a test branch of yours on Vercel. Let me know how you get on.

**Reply by gajraj-gan** [reply]
Thanks @bitttttten ! This looks good, I will test it :)

**Reply by gajraj-gan** [reply]
hey @bitttttten ! I have tried your plugin, unfortunately my CSS isnt loading anymore lol

**Reply by gajraj-gan** [reply]
I am not sure why it happened, and how it works in my normal build without this plugin. This is happening on CSS/SCSS modules btw, have you used them in your app?

**Reply by gajraj-gan** [reply]
One thing that I however noticed that, in a normal build (/without this) we have this variable `const __vite__fileDeps` in the .js files. But not here. Could be related

**Reply by gajraj-gan** [reply]
I kinda wrote something of my own and it works ig, but `import react-loading-skeleton/dist/skeleton.css` doesnt work. Weird. I'll have to study this more lol.

It was importing it as 'skeleton.js' for some reason in the build

**Reply by bitttttten** [reply]
@gajraj-gan can u provide a repo to reproduce? then i can look into it. we were importing css from all over the place and it was working. it may be that your html file is a lil different? happy to help if i can get into the source code to see what's up

[a comment by an automated account is not shown]
